### ISSUE TITLE
Documentation info

### ISSUE THREAD

**Issue by NicoleVenachi** [a2134666/p4_wecmp#1]
I wanted to ask you about the Documentation of your code, i didn't get some parts of your p4 code, and it seems like the Readme info disagree with your WECMP project. I would really aprecciate if you could please explain a little more about testing the code to certainly know which path is taking every message. Morveover, i wanted to ask why you are using two .p4 files, and two topology files. 

I'm sorry to bother you, and i would really apprecciate your help, because it could helps me in an investigation project.

Thank you for your time and consideration.

**Comment by NicoleVenachi** [comment]
Also, i forget to ask if this was the Paper of this Repository, https://ieeexplore.ieee.org/document/8549549?

**Comment by zml25904** [comment]
Sorry, this is my homework a year ago. I almost forget how it work and its result. I remember I combined several example from p4 tutorial to reproduce that paper you attached. So, the documentation info is copied from one of the p4 tutorial exercise which was used as prototype to start. I have the vm file for this work, if you want, maybe I can send it to you.

**Comment by jmcc153** [comment]
Thank you for your answer, Do you perhaps know what load_balance.p4.sw is used for? Because in the p4's tutorials they used a only p4 file

**Comment by NicoleVenachi** [comment]
Thanks for your

> Sorry, this is my homework a year ago. I almost forget how it work and its result. I remember I combined several example from p4 tutorial to reproduce that paper you attached. So, the documentation info is copied from one of the p4 tutorial exercise which was used as prototype to start. I have the vm file for this work, if you want, maybe I can send it to you.

Thanks for your repply. Ok, i understand the situation. Yeah, it could be really useful for us to have your VM, i would really appreaciate if you could shared it with us. Nevertheless, our main doubt is why do you used two '.p4' files, please if do you remembered it, let us know it, it could be the really helpful for us.

**Comment by zml25904** [comment]
Maybe it is just a backup, because I did not use it in the final testing. I have no idea why I named it ".sw" not other term related to "backup".
[
![擷取](https://user-images.githubusercontent.com/30485602/158513243-ba919407-c28d-4c61-9eb3-15273bbd907e.JPG)
](url)

**Comment by NicoleVenachi** [comment]
> Maybe it is just a backup, because I did not use it in the final testing. I have no idea why I named it ".sw" not other term related to "backup". [ ![擷取](https://user-images.githubusercontent.com/30485602/158513243-ba919407-c28d-4c61-9eb3-15273bbd907e.JPG) ](url)

I see, may that's the reason 'cause we've run the project without this file (removing it also from the controller) and it seems to works fine. Thanks a lot!!! 

However if you have the VM or the files used in the final testing, they could be really useful too! We would really appreciate if you could shared them with us.

**Comment by NicoleVenachi** [comment]
I see, may that's the reason 'cause we've run the project without this file (removing it also from the controller) and it seems to works fine. Thanks a lot!!!

However if you have the VM or the files used in the final testing, they could be really useful too! We would really appreciate if you could shared them with us.

**Comment by zml25904** [comment]
OK, may you give me your mail, or somewhere that I can upload. The vm file costs 5.19GB which my G suite of school can't support.

**Comment by jmcc153** [comment]
Ok, ludwing@unicauca.edu.co please, thank you !!

**Comment by leonSjp** [comment]
I also want to get the vm or  the files used in the final testing ，and I wanted to ask  if the load_balance.p4 is the final test file in your project，980937713@qq.com is my email，thanks a lot！！ i would really apprecciate your help！
Thank you for your time and consideration.

**Comment by zml25904** [comment]
@leonSjp You are right, the "load_balance.p4" is the final test file. I just update the README, you could find all you need.

**Comment by NicoleVenachi** [comment]
@leonSjp Me and my partner @jmcc153 have also implemented a full solution from this repository, you might found it useful. I attach the link https://github.com/NicoleVenachi/p4_probebased_loadbalancing, also we've write a paper though we've never published it https://drive.google.com/file/d/1CHKw-YTRCEayEwu3Yn2WOlu_7SemcpIN/view?usp=share_link
Once again, thanks for your help @a2134666

**Comment by zml25904** [comment]
@NicoleVenachi You're welcome, it's my pleasure. Would you mind if I put your repo in the README.

**Comment by NicoleVenachi** [comment]
@a2134666 Sure, no problem

**Comment by leonSjp** [comment]
> 
Thanks a lot！It really helps me！ i would really apprecciate your help！

**Comment by leonSjp** [comment]
@NicoleVenachi Thank you very much！It really helps me a lot！
